include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -1418,26 +1421,7 @@ template<int dim, typename Number>
 void
 SpatialOperatorBase<dim, Number>::update_viscosity(VectorType const & velocity) const
 {
-  AssertThrow(param.viscous_problem() and param.viscosity_is_variable(),
-              dealii::ExcMessage(
-                "Updating viscosity reasonable for variable viscosity models only."));
-
-  // reset the viscosity stored
-  // viscosity = viscosity_newtonian_limit
-  viscous_kernel->set_constant_coefficient(viscous_kernel_data.viscosity);
-
-  // add contribution from generalized Newtonian model
-  // viscosity += generalized_newtonian_viscosity(viscosity_newtonian_limit)
-  if(param.generalized_newtonian_model_data.is_active)
-    generalized_newtonian_model.add_viscosity(velocity);
-
-  // add contribution from turbulence model
-  // viscosity += turbulent_viscosity(viscosity)
-  // note that the apparent viscosity is used to compute the turbulent viscosity, such that the
-  // *sequence of calls matters*, i.e., we can only compute the turbulent viscosity once the laminar
-  // viscosity has been computed
-  if(param.turbulence_model_data.is_active)
-    turbulence_model.add_viscosity(velocity);
+  momentum_operator.update_viscosity(velocity);
```

Comment:
before, the variable viscosity models where in `SpatialOperatorBase`.
the current version calls then 'MomentumOperatorBase::update_viscosity()' (this function is needed to update the viscosity in MG)
...
both not ideal given the necessity to now also call from within MG.
I would like to hand the variable viscosity models to the `viscous_kernel` and call `viscous_kernel.update_viscosity(velocity)`.